apps/coordinator/ts/health/health.service.ts

```diff
@@ -0,0 +1,176 @@
+import { EMode } from "@maci-protocol/sdk";
+import { Injectable } from "@nestjs/common";
+import { formatEther } from "ethers";
+import { zeroAddress } from "viem";
+
+import { constants, statSync, promises } from "fs";
+import path from "path";
+
+import type { ICheckRapidsnark, ICheckWalletFunds, ICheckZkeysDirectory, IHealthCheckResponse } from "./types";
+
+import { ESupportedNetworks, getSigner } from "../common";
+import { FileService } from "../file/file.service";
+
+/**
+ * Health service functions to check the status of the coordinator service.
+ */
+@Injectable()
+export class HealthService {
+  /**
+   * Create a new instance of the HealthService.
+   *
+   * @param fileService - file service
+   */
+  constructor(private readonly fileService: FileService) {}
+
+  /**
+   * Check if rapidsnark is available and executable.
+   * @returns the rapidsnark executable path and its executability status
+   */
+  async checkRapidsnark(): Promise<ICheckRapidsnark> {
+    const rapidsnarkExecutablePath = process.env.COORDINATOR_RAPIDSNARK_EXE || "";
+
+    const rapidsnarkIsAccessible = await promises
+      // eslint-disable-next-line no-bitwise
+      .access(rapidsnarkExecutablePath, constants.F_OK | constants.X_OK)
+      .then(() => true)
+      .catch(() => false);
+
+    const rapidsnarkIsExecutable = await promises
+      // eslint-disable-next-line no-bitwise
+      .stat(rapidsnarkExecutablePath)
+      .then((stats) => stats.isFile())
+      .catch(() => false);
+
+    return {
+      rapidsnarkExecutablePath,
+      rapidsnarkIsAccessible,
+      rapidsnarkIsExecutable,
+    };
+  }
+
+  /**
+   * Check if zkeys directory exists
+   * @returns the existence of the zkeys directory and the required zkeys
+   */
+  checkZkeysDirectory(): ICheckZkeysDirectory {
+    const zkeysDirectoryPath = path.resolve(process.env.COORDINATOR_ZKEY_PATH!);
+    const requiredZkeyNames = [
+      {
+        name: process.env.COORDINATOR_TALLY_ZKEY_NAME,
+        modes: [EMode.QV, EMode.NON_QV],
+      },
+      {
+        name: process.env.COORDINATOR_MESSAGE_PROCESS_ZKEY_NAME,
+        modes: [EMode.QV, EMode.NON_QV, EMode.FULL],
+      },
+      {
+        name: process.env.COORDINATOR_POLL_JOINING_ZKEY_NAME,
+      },
+      {
+        name: process.env.COORDINATOR_POLL_JOINED_ZKEY_NAME,
+      },
+    ];
+
+    let zkeysDirectoryExists = false;
+
+    try {
+      const stats = statSync(zkeysDirectoryPath);
+      zkeysDirectoryExists = stats.isDirectory();
+    } catch (error) {
+      zkeysDirectoryExists = false;
+    }
+
+    const availableZkeys = requiredZkeyNames.reduce((acc: ICheckZkeysDirectory["availableZkeys"], { name, modes }) => {
+      if (!modes) {
+        try {
+          acc[name!] = this.fileService.getZkeyFilePaths(name!);
+        } catch (error) {
+          acc[name!] = undefined;
+        }
+
+        return acc;
+      }
+
+      const modeEntries = modes.reduce((modeAcc: ICheckZkeysDirectory["availableZkeys"], mode) => {
```

Comment:
And you don't need to disable eslint inside reduce
```suggestion
      const modeEntries = modes.reduce<ICheckZkeysDirectory["availableZkeys"]>((modeAcc, mode) => {
```

Reply:
Anton in my computer the eslint error is not desappearing with this solution:

![image](https://github.com/user-attachments/assets/cdc19f2c-6829-44df-b09f-531fba96f87f)


Please let's merge this PR asap because I need to focus on other issues

Reply:
@0xmad I just squashed the commits and activated the auto merge! 🔥 I need your approval in the requested changes